src/main/java/it/gov/pagopa/pu/bff/service/debt_position/DebtPositionRetrieverServiceImpl.java

```diff
@@ -99,4 +116,40 @@ public boolean deleteDebtPosition(Long organizationId, Long debtPositionId, User
     return debtPositionService.deleteDebtPosition(debtPositionId,accessToken);
   }
 
+  @Override
+  public Resource getDebtPositionNoticesZip(Long organizationId, Long debtPositionId, UserInfo loggedUser, String accessToken) {
+    AuthorizationService.validateUserForOrganizationId(organizationId, loggedUser);
+
+    DebtPositionDTO debtPosition = debtPositionService.getDebtPosition(debtPositionId, accessToken);
+    if (debtPosition == null){
+      return null;
+    }
+
+    List<FileResourceDTO> pdfResources = debtPosition
+      .getPaymentOptions()
+      .stream()
+      .flatMap(po ->
+        po.getInstallments()
+          .stream()
+          .filter(
+            i -> i.getStatus() != null &&
```

Comment:
this check is redundant, it cannot be null and equals to one of the 2 values
```suggestion
            i -> 
```